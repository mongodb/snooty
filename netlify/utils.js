import fs from 'fs';
import path from 'path';

/**
 * Parses build hook data from the expected temporary location
 * @returns {object | undefined}
 */
const parseBuildHookData = () => {
  // This file does not currently exist, but should be created on Netlify builds as part of `npm run build:netlify`.
  // The INCOMING_HOOK_BODY env var is not automatically passed along to functions, so we use a txt file to save it
  const relativeFilePath = '../build-hook.txt';
  const buildHookDataString = fs.readFileSync(path.resolve(__dirname, relativeFilePath), 'utf-8');
  if (!buildHookDataString) {
    console.log('No build hook data found.');
    return;
  }
  return JSON.parse(buildHookDataString);
};

export const constructResPayload = (event) => {
  const buildHookData = parseBuildHookData();
  const parsedEventBody = JSON.parse(event.body);
  // This is Netlify's default post-deployment payload. We include it with our custom data in case
  // we want to process any information
  const netlifyPayload = parsedEventBody.payload;
  return {
    netlifyPayload,
    ...buildHookData,
  };
};
