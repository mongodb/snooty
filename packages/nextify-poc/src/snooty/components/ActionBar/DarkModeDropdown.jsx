import React, { useCallback, useContext, useRef, useState } from 'react';
import { cx, css } from '@leafygreen-ui/emotion';
import Icon from '@leafygreen-ui/icon';
import IconButton from '@leafygreen-ui/icon-button';
import { Menu, MenuItem } from '@leafygreen-ui/menu';
import { DarkModeContext } from '../../context/dark-mode-context';
import { theme } from '../../theme/docsTheme';
import { reportAnalytics } from '../../utils/report-analytics';
import IconDarkmode from '../icons/DarkMode';

const iconStyling = css`
  display: block;
  align-content: center;
  cursor: pointer;

  > div {
    position: relative;
  }
`;

const menuStyling = css`
  width: fit-content;
  margin-top: ${theme.size.small};
`;

const DROPDOWN_ICON_SIZE = 20;
const darkModeSvgStyle = {
  width: DROPDOWN_ICON_SIZE,
  height: DROPDOWN_ICON_SIZE,
};

const DarkModeDropdown = () => {
  const dropdownRef = useRef();

  // not using dark mode from LG/provider here to account for case of 'system' dark theme
  const { setDarkModePref, darkModePref } = useContext(DarkModeContext);

  const [open, setOpen] = useState(false);

  const select = useCallback(
    (selectedPref) => {
      reportAnalytics('DarkModeSelection', {
        value: selectedPref,
      });
      setDarkModePref(selectedPref);
      setOpen(false);
    },
    [setDarkModePref, setOpen]
  );

  return (
    <div
      className={cx(
        css`
          position: relative;
        `
      )}
      ref={dropdownRef}
    >
      <Menu
        className={cx(menuStyling)}
        portalContainer={dropdownRef.current}
        scrollContainer={dropdownRef.current}
        justify={'start'}
        align={'bottom'}
        open={open}
        setOpen={() => {
          reportAnalytics('DarkModeMenu', {
            action: open ? 'closed' : 'opened',
          });
          setOpen((e) => !e);
        }}
        trigger={
          <IconButton className={cx(iconStyling)} aria-label="Dark Mode Menu" aria-labelledby="Dark Mode Menu">
            {darkModePref === 'system' ? (
              <IconDarkmode />
            ) : (
              <Icon size={24} glyph={darkModePref === 'dark-theme' ? 'Moon' : 'Sun'} />
            )}
          </IconButton>
        }
      >
        <MenuItem
          active={darkModePref === 'light-theme'}
          onClick={() => select('light-theme')}
          glyph={<Icon size={DROPDOWN_ICON_SIZE} glyph={'Sun'} />}
        >
          Light
        </MenuItem>
        <MenuItem
          active={darkModePref === 'dark-theme'}
          onClick={() => select('dark-theme')}
          glyph={<Icon size={DROPDOWN_ICON_SIZE} glyph={'Moon'} />}
        >
          Dark
        </MenuItem>
        <MenuItem
          active={darkModePref === 'system'}
          onClick={() => select('system')}
          glyph={
            <IconDarkmode
              className={css`
                svg {
                  margin-right: ${theme.size.default};
                }
              `}
              styles={darkModeSvgStyle}
            />
          }
        >
          System
        </MenuItem>
      </Menu>
    </div>
  );
};

DarkModeDropdown.propTypes = {};

export default DarkModeDropdown;
