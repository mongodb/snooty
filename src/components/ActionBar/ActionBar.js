import React from 'react';
import styled from '@emotion/styled';
import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
import { palette } from '@leafygreen-ui/palette';
import { theme } from '../../theme/docsTheme';
import ChatbotUi from '../ChatbotUi';
import DarkModeDropdown from './DarkModeDropdown';

const ActionBarContainer = styled('div')`
  display: flex;
  justify-content: space-between;
  padding-top: ${theme.size.small};
  padding-bottom: ${theme.size.small};
  padding-right: ${theme.size.large};
  width: 100%;
  position: sticky;
  top: 0;
  flex-wrap: wrap;
  z-index: ${theme.zIndexes.actionBar};
  background-color: ${(props) => (props.darkMode ? palette.black : palette.white)};
  border-bottom: 1px solid ${(props) => (props.darkMode ? palette.gray.dark2 : palette.gray.light2)};

  @media ${theme.screenSize.mediumAndUp} {
    & > div {
      flex: 0 1 auto;
    }
  }

  @media ${theme.screenSize.upToMedium} {
    justify-content: space-between;
    padding-right: 0;
  }
`;

const ActionBarSearchContainer = styled.div`
  align-items: center;
  display: flex;
  width: 80%;

  @media ${theme.screenSize.upToMedium} {
    width: 100%;
  }

  @media ${theme.screenSize.upToSmall} {
    & > div {
      padding: ${theme.size.default} 32px;
    }
  }
`;

const ActionsBox = styled('div')`
  display: flex;
  align-items: center;
  column-gap: ${theme.size.default};

  @media ${theme.screenSize.upToMedium} {
    padding-left: 3rem;
  }

  @media ${theme.screenSize.upToSmall} {
    padding-left: 2rem;
  }
`;

// Note: When working on this component further, please check with design on how it should look in the errorpage template (404) as well!
const ActionBar = ({ template, ...props }) => {
  const { darkMode } = useDarkMode();
  return (
    <ActionBarContainer className={props.className} darkMode={darkMode}>
      <ActionBarSearchContainer>
        <ChatbotUi darkMode={darkMode} />
      </ActionBarSearchContainer>
      <ActionsBox>
        {template !== 'openapi' && <DarkModeDropdown />}
        {template !== 'errorpage' && (
          <div>
            <button>Feedback</button>
          </div>
        )}
      </ActionsBox>
    </ActionBarContainer>
  );
};

ActionBar.propTypes = {};

export default ActionBar;
