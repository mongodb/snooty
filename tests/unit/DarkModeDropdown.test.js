import React from 'react';
import { render, act } from '@testing-library/react';
import LeafyGreenProvider from '@leafygreen-ui/leafygreen-provider';
import { DarkModeContext } from '../../src/context/dark-mode-context';
import * as snootyMetadata from '../../src/utils/use-snooty-metadata';
import * as MediaHooks from '../../src/hooks/use-media';
import DarkModeDropdown from '../../src/components/ActionBar/DarkModeDropdown';

import { setDesktop } from '../utils';
import { DARK_MODE_ANNOUNCED } from '../../src/components/ActionBar/DarkModeGuideCue';

let darkModePref = 'light-theme';

const setDarkModePref = jest.fn((value) => {
  darkModePref = value;
});

// mock useSnootyMetadata
jest.spyOn(snootyMetadata, 'default').mockImplementation(() => ({
  project: 'landing',
}));

// mock useMedia
jest.spyOn(MediaHooks, 'default').mockImplementation(() => ({}));

// mock window.localStorage
const storage = {
  [DARK_MODE_ANNOUNCED]: 'true',
};
jest.spyOn(window.localStorage.__proto__, 'setItem').mockImplementation((key, value) => {
  storage[key] = value;
});

const mountDarkModeDropdown = () => {
  return render(
    <DarkModeContext.Provider value={{ setDarkModePref, darkModePref }}>
      <LeafyGreenProvider baseFontSize={16} darkMode={darkModePref === 'dark-theme'}>
        <DarkModeDropdown />
      </LeafyGreenProvider>
    </DarkModeContext.Provider>
  );
};

describe('DarkMode Dropdown component', () => {
  beforeEach(setDesktop);

  it('renders dark mode dropdown', async () => {
    // first snapshot of closed menu
    const elm = mountDarkModeDropdown();
    expect(elm.asFragment()).toMatchSnapshot();
    const button = await elm.findByLabelText('Dark Mode Menu');
    await act(async () => {
      button.click();
    });
    // second snapshot of open menu
    expect(elm.asFragment()).toMatchSnapshot();
  });

  // test it changes to system and dark mode
  it('updates dark mode when selecting a different option', async () => {
    const elm = mountDarkModeDropdown();
    const button = await elm.findByLabelText('Dark Mode Menu');
    await act(async () => {
      button.click();
    });
    const darkModeSelections = await elm.findAllByRole('menuitem');
    await act(async () => {
      darkModeSelections[1].click();
    });
    expect(darkModePref).toBe('dark-theme');
    const darkElm = mountDarkModeDropdown();
    expect(darkElm.asFragment()).toMatchSnapshot();
  });
});
