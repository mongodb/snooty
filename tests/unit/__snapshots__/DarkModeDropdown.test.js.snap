// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkMode Dropdown component renders dark mode dropdown 1`] = `
<DocumentFragment>
  .emotion-0 {
  border: none;
  -webkit-appearance: unset;
  padding: unset;
  display: inline-block;
  border-radius: 100px;
  color: #89979B;
  position: relative;
  cursor: pointer;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: color 150ms ease-in-out;
  transition: color 150ms ease-in-out;
  background-color: rgba(255, 255, 255, 0);
  height: 28px;
  width: 28px;
  cursor: pointer;
}

.emotion-0:before {
  content: '';
  -webkit-transition: 150ms all ease-in-out;
  transition: 150ms all ease-in-out;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  border-radius: 100%;
  opacity: 0;
  -webkit-transform: scale(0.8);
  -moz-transform: scale(0.8);
  -ms-transform: scale(0.8);
  transform: scale(0.8);
}

.emotion-0:active:before,
.emotion-0:hover:before,
.emotion-0:focus:before {
  opacity: 1;
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:active,
.emotion-0:hover {
  color: #3D4F58;
}

.emotion-0:active:before,
.emotion-0:hover:before {
  background-color: #E7EEEC;
}

.emotion-0:focus {
  color: #1A567E;
}

.emotion-0:focus:before {
  background-color: #C5E4F2;
}

.emotion-1 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

<div
    aria-disabled="false"
    aria-expanded="false"
    aria-haspopup="true"
    aria-label="Dark Mode Menu"
    aria-labelledby="Dark Mode Menu"
    class="emotion-0"
    tabindex="0"
  >
    <div
      class="emotion-1"
    >
      <svg
        aria-label="Sun Icon"
        class=""
        fill="none"
        height="16"
        role="img"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.24 10.83a.996.996 0 1 0-1.41 1.41l.71.71a.996.996 0 1 0 1.41-1.41l-.71-.71ZM8 12c-.55 0-1 .45-1 1v1c0 .55.45 1 1 1s1-.45 1-1v-1c0-.55-.45-1-1-1ZM8 11a3 3 0 1 0 0-6 3 3 0 0 0 0 6ZM14 7h-1c-.55 0-1 .45-1 1s.45 1 1 1h1c.55 0 1-.45 1-1s-.45-1-1-1ZM3.76 10.83l-.71.71a.996.996 0 1 0 1.41 1.41l.71-.71a.996.996 0 1 0-1.41-1.41ZM8 4c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1v1c0 .55.45 1 1 1ZM3.76 5.17a.996.996 0 1 0 1.41-1.41l-.71-.71a.996.996 0 1 0-1.41 1.41l.71.71ZM4 8c0-.55-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1h1c.55 0 1-.45 1-1ZM12.24 5.17l.71-.71a.996.996 0 1 0-1.41-1.41l-.71.71a.996.996 0 1 0 1.41 1.41Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DarkMode Dropdown component renders dark mode dropdown 2`] = `
<DocumentFragment>
  .emotion-0 {
  border: none;
  -webkit-appearance: unset;
  padding: unset;
  display: inline-block;
  border-radius: 100px;
  color: #89979B;
  position: relative;
  cursor: pointer;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: color 150ms ease-in-out;
  transition: color 150ms ease-in-out;
  background-color: rgba(255, 255, 255, 0);
  height: 28px;
  width: 28px;
  cursor: pointer;
}

.emotion-0:before {
  content: '';
  -webkit-transition: 150ms all ease-in-out;
  transition: 150ms all ease-in-out;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  border-radius: 100%;
  opacity: 0;
  -webkit-transform: scale(0.8);
  -moz-transform: scale(0.8);
  -ms-transform: scale(0.8);
  transform: scale(0.8);
}

.emotion-0:active:before,
.emotion-0:hover:before,
.emotion-0:focus:before {
  opacity: 1;
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:active,
.emotion-0:hover {
  color: #3D4F58;
}

.emotion-0:active:before,
.emotion-0:hover:before {
  background-color: #E7EEEC;
}

.emotion-0:focus {
  color: #1A567E;
}

.emotion-0:focus:before {
  background-color: #C5E4F2;
}

.emotion-1 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-2 {
  display: none;
}

.emotion-3 {
  position: absolute;
  -webkit-transition: -webkit-transform 150ms ease-in-out,opacity 150ms ease-in-out;
  transition: transform 150ms ease-in-out,opacity 150ms ease-in-out;
  opacity: 0;
  top: calc(100% + 6px);
  left: 0;
  transform-origin: left top;
  -webkit-transform: translate3d(0, -6px, 0) scale(0.8);
  -moz-transform: translate3d(0, -6px, 0) scale(0.8);
  -ms-transform: translate3d(0, -6px, 0) scale(0.8);
  transform: translate3d(0, -6px, 0) scale(0.8);
}

.emotion-4 {
  width: 210px;
  border-radius: 12px;
  overflow: auto;
  padding: 14px 0;
  background-color: #001E2B;
  border: 1px solid #001E2B;
  max-height: 344px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  margin-top: 8px;
}

.emotion-5 {
  overflow: auto;
  list-style: none;
  margin-block-start: 0px;
  margin-block-end: 0px;
  -webkit-padding-start: 0px;
  padding-inline-start: 0px;
  padding: 0px;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  box-sizing: border-box;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  margin: unset;
  padding-left: 20px;
  padding-right: 20px;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 13px;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  border: none;
  -webkit-transition: background-color 150ms ease-in-out;
  transition: background-color 150ms ease-in-out;
  color: #FFFFFF;
  background-color: #001E2B;
  min-height: 36px;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #001E2B;
}

.emotion-6:focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-6:before {
  content: '';
  position: absolute;
  width: 4px;
  left: 0px;
  border-radius: 0 4px 4px 0;
  background-color: transparent;
  -webkit-transition: background-color 150ms ease-in-out;
  transition: background-color 150ms ease-in-out;
}

.emotion-6:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-6:hover,
.emotion-6:active {
  background-color: #1C2D38;
}

.emotion-6:before {
  height: 22px;
}

.emotion-6:before {
  background-color: #00ED64;
}

.emotion-6:hover {
  color: #00ED64;
}

.emotion-6:hover:before {
  background-color: #00ED64;
}

.emotion-6:focus-visible {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #0C2657;
  color: #FFFFFF;
}

.emotion-6:focus-visible:before {
  background-color: #0498EC;
}

.emotion-6::-moz-focus-inner {
  border: 0;
}

.emotion-7 {
  margin-right: 16px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: #889397;
  color: #00ED64;
}

.lg-ui-menu-item-container-0000:focus-visible .emotion-7 {
  color: #E1F7FF;
}

.emotion-8 {
  width: 100%;
  overflow: hidden;
  padding: 2px 0;
}

.emotion-9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
  font-size: 13px;
  font-weight: 500;
  font-weight: 700;
  color: #00ED64;
}

.emotion-9:after {
  content: attr(data-text);
  height: 0;
  font-weight: 700;
  visibility: hidden;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
}

.emotion-9 .lg-ui-menu-item-container-0000:not(:disabled):hover .emotion-9 .lg-ui-menu-item-container-0000 {
  font-weight: 700;
}

.emotion-9 .lg-ui-menu-item-container-0000:not(:disabled):hover .emotion-9 .lg-ui-menu-item-container-0000 {
  color: #00ED64;
}

.lg-ui-menu-item-container-0000:focus-visible .emotion-9 {
  color: #E1F7FF;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  box-sizing: border-box;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  margin: unset;
  padding-left: 20px;
  padding-right: 20px;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 13px;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  border: none;
  -webkit-transition: background-color 150ms ease-in-out;
  transition: background-color 150ms ease-in-out;
  color: #FFFFFF;
  background-color: #001E2B;
  min-height: 36px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-10:focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-10:before {
  content: '';
  position: absolute;
  width: 4px;
  left: 0px;
  border-radius: 0 4px 4px 0;
  background-color: transparent;
  -webkit-transition: background-color 150ms ease-in-out;
  transition: background-color 150ms ease-in-out;
}

.emotion-10:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-10:hover,
.emotion-10:active {
  background-color: #1C2D38;
}

.emotion-10:before {
  height: 22px;
}

.emotion-10:focus-visible {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #0C2657;
  color: #FFFFFF;
}

.emotion-10:focus-visible:before {
  background-color: #0498EC;
}

.emotion-10::-moz-focus-inner {
  border: 0;
}

.emotion-11 {
  margin-right: 16px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: #889397;
}

.lg-ui-menu-item-container-0000:focus-visible .emotion-11 {
  color: #E1F7FF;
}

.emotion-13 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
  font-size: 13px;
  font-weight: 500;
}

.emotion-13:after {
  content: attr(data-text);
  height: 0;
  font-weight: 700;
  visibility: hidden;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
}

.emotion-13 .lg-ui-menu-item-container-0000:not(:disabled):hover .emotion-13 .lg-ui-menu-item-container-0000 {
  font-weight: 700;
}

.lg-ui-menu-item-container-0000:focus-visible .emotion-13 {
  color: #E1F7FF;
}

<div
    aria-disabled="false"
    aria-expanded="true"
    aria-haspopup="true"
    aria-label="Dark Mode Menu"
    aria-labelledby="Dark Mode Menu"
    class="emotion-0"
    tabindex="0"
  >
    <div
      class="emotion-1"
    >
      <svg
        aria-label="Sun Icon"
        class=""
        fill="none"
        height="16"
        role="img"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.24 10.83a.996.996 0 1 0-1.41 1.41l.71.71a.996.996 0 1 0 1.41-1.41l-.71-.71ZM8 12c-.55 0-1 .45-1 1v1c0 .55.45 1 1 1s1-.45 1-1v-1c0-.55-.45-1-1-1ZM8 11a3 3 0 1 0 0-6 3 3 0 0 0 0 6ZM14 7h-1c-.55 0-1 .45-1 1s.45 1 1 1h1c.55 0 1-.45 1-1s-.45-1-1-1ZM3.76 10.83l-.71.71a.996.996 0 1 0 1.41 1.41l.71-.71a.996.996 0 1 0-1.41-1.41ZM8 4c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1v1c0 .55.45 1 1 1ZM3.76 5.17a.996.996 0 1 0 1.41-1.41l-.71-.71a.996.996 0 1 0-1.41 1.41l.71.71ZM4 8c0-.55-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1h1c.55 0 1-.45 1-1ZM12.24 5.17l.71-.71a.996.996 0 1 0-1.41-1.41l-.71.71a.996.996 0 1 0 1.41 1.41Z"
          fill="currentColor"
        />
      </svg>
      <span
        class="emotion-2"
      />
      <div
        class="emotion-3"
      >
        <div
          class="lg-ui-popover-content-0000"
        >
          <div
            class="emotion-4"
          >
            <ul
              class="emotion-5"
              role="menu"
            >
              <li
                role="none"
              >
                <button
                  aria-current="true"
                  aria-disabled="false"
                  class="lg-ui-menu-item-container-0000 emotion-6"
                  role="menuitem"
                  tabindex="-1"
                >
                  <svg
                    alt=""
                    aria-hidden="true"
                    class="emotion-7"
                    fill="none"
                    height="16"
                    role="presentation"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12.24 10.83a.996.996 0 1 0-1.41 1.41l.71.71a.996.996 0 1 0 1.41-1.41l-.71-.71ZM8 12c-.55 0-1 .45-1 1v1c0 .55.45 1 1 1s1-.45 1-1v-1c0-.55-.45-1-1-1ZM8 11a3 3 0 1 0 0-6 3 3 0 0 0 0 6ZM14 7h-1c-.55 0-1 .45-1 1s.45 1 1 1h1c.55 0 1-.45 1-1s-.45-1-1-1ZM3.76 10.83l-.71.71a.996.996 0 1 0 1.41 1.41l.71-.71a.996.996 0 1 0-1.41-1.41ZM8 4c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1v1c0 .55.45 1 1 1ZM3.76 5.17a.996.996 0 1 0 1.41-1.41l-.71-.71a.996.996 0 1 0-1.41 1.41l.71.71ZM4 8c0-.55-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1h1c.55 0 1-.45 1-1ZM12.24 5.17l.71-.71a.996.996 0 1 0-1.41-1.41l-.71.71a.996.996 0 1 0 1.41 1.41Z"
                      fill="currentColor"
                    />
                  </svg>
                  <div
                    class="emotion-8"
                  >
                    <div
                      class="emotion-9"
                      data-text="Light"
                    >
                      Light
                    </div>
                  </div>
                </button>
              </li>
              <li
                role="none"
              >
                <button
                  aria-current="false"
                  aria-disabled="false"
                  class="lg-ui-menu-item-container-0000 emotion-10"
                  role="menuitem"
                  tabindex="-1"
                >
                  <svg
                    alt=""
                    aria-hidden="true"
                    class="emotion-11"
                    fill="none"
                    height="16"
                    role="presentation"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13.787 10.074a.713.713 0 0 0-.864-.116c-.125.077-.26.135-.394.192-1.766.72-3.888-.21-4.704-2.208-.73-1.795-.086-3.734 1.344-4.617a.72.72 0 0 0 .317-.807A.709.709 0 0 0 8.795 2H8.67c-3.312 0-6 2.688-6 6s2.688 6 6 6a6.003 6.003 0 0 0 5.232-3.062.715.715 0 0 0-.125-.864h.01Z"
                      fill="currentColor"
                    />
                  </svg>
                  <div
                    class="emotion-8"
                  >
                    <div
                      class="emotion-13"
                      data-text="Dark"
                    >
                      Dark
                    </div>
                  </div>
                </button>
              </li>
              <li
                role="none"
              >
                <button
                  aria-current="false"
                  aria-disabled="false"
                  class="lg-ui-menu-item-container-0000 emotion-10"
                  role="menuitem"
                  tabindex="-1"
                >
                  <svg
                    class="emotion-11"
                    fill="none"
                    height="20"
                    role="presentation"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <mask
                      height="20"
                      id="mask0_2936_7195"
                      maskUnits="userSpaceOnUse"
                      width="20"
                      x="0"
                      y="0"
                    >
                      <rect
                        fill="#D9D9D9"
                        height="20"
                        width="20"
                      />
                    </mask>
                    <g
                      mask="url(#mask0_2936_7195)"
                    >
                      <path
                        d="M3.49677 14.9433C3.00064 14.9433 2.57799 14.7687 2.22883 14.4196C1.87966 14.0704 1.70508 13.6478 1.70508 13.1516V4.30927C1.70508 3.81314 1.87966 3.39049 2.22883 3.04133C2.57799 2.69216 3.00064 2.51758 3.49677 2.51758H16.5022C16.9983 2.51758 17.421 2.69216 17.7701 3.04133C18.1193 3.39049 18.2939 3.81314 18.2939 4.30927V13.1516C18.2939 13.6478 18.1193 14.0704 17.7701 14.4196C17.421 14.7687 16.9983 14.9433 16.5022 14.9433H13.937L14.6068 15.6181C14.6932 15.7038 14.7579 15.8001 14.8011 15.9072C14.8443 16.0142 14.8659 16.1268 14.8659 16.245V16.6304C14.8659 16.8654 14.7828 17.066 14.6167 17.2321C14.4505 17.3983 14.2499 17.4814 14.0149 17.4814H5.98408C5.74904 17.4814 5.54844 17.3983 5.38229 17.2321C5.21615 17.066 5.13308 16.8654 5.13308 16.6304V16.245C5.13308 16.1268 5.15467 16.0142 5.19785 15.9072C5.24102 15.8001 5.30578 15.7038 5.39214 15.6181L6.06197 14.9433H3.49677ZM3.49677 13.1516H16.5022V4.30927H3.49677V13.1516Z"
                        fill="#889397"
                      />
                    </g>
                  </svg>
                  <div
                    class="emotion-8"
                  >
                    <div
                      class="emotion-13"
                      data-text="System"
                    >
                      System
                    </div>
                  </div>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DarkMode Dropdown component updates dark mode when selecting a different option 1`] = `
<DocumentFragment>
  .emotion-0 {
  border: none;
  -webkit-appearance: unset;
  padding: unset;
  display: inline-block;
  border-radius: 100px;
  color: #89979B;
  position: relative;
  cursor: pointer;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: color 150ms ease-in-out;
  transition: color 150ms ease-in-out;
  background-color: rgba(255, 255, 255, 0);
  height: 28px;
  width: 28px;
  cursor: pointer;
}

.emotion-0:before {
  content: '';
  -webkit-transition: 150ms all ease-in-out;
  transition: 150ms all ease-in-out;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  border-radius: 100%;
  opacity: 0;
  -webkit-transform: scale(0.8);
  -moz-transform: scale(0.8);
  -ms-transform: scale(0.8);
  transform: scale(0.8);
}

.emotion-0:active:before,
.emotion-0:hover:before,
.emotion-0:focus:before {
  opacity: 1;
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:active,
.emotion-0:hover {
  color: #3D4F58;
}

.emotion-0:active:before,
.emotion-0:hover:before {
  background-color: #E7EEEC;
}

.emotion-0:focus {
  color: #1A567E;
}

.emotion-0:focus:before {
  background-color: #C5E4F2;
}

.emotion-1 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

<div
    aria-disabled="false"
    aria-expanded="false"
    aria-haspopup="true"
    aria-label="Dark Mode Menu"
    aria-labelledby="Dark Mode Menu"
    class="emotion-0"
    tabindex="0"
  >
    <div
      class="emotion-1"
    >
      <svg
        aria-label="Moon Icon"
        class=""
        fill="none"
        height="16"
        role="img"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.787 10.074a.713.713 0 0 0-.864-.116c-.125.077-.26.135-.394.192-1.766.72-3.888-.21-4.704-2.208-.73-1.795-.086-3.734 1.344-4.617a.72.72 0 0 0 .317-.807A.709.709 0 0 0 8.795 2H8.67c-3.312 0-6 2.688-6 6s2.688 6 6 6a6.003 6.003 0 0 0 5.232-3.062.715.715 0 0 0-.125-.864h.01Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
